src/task-actions.ts

```diff
@@ -339,31 +339,44 @@ async function handleContractVerification(
     const paths = [mainPath];
 
     const bodyFormData = new FormData();
-    bodyFormData.append("contract-name", path.parse(mainPath).base);
+    bodyFormData.append("compiler-version", args.compilerVersion);
+    bodyFormData.append("license", args.license || "No License (None)");
+    bodyFormData.append("account-contract", args.accountContract || "false");
 
     // Dependencies (non-main contracts) are in args.paths
     if (args.paths) {
         paths.push(...args.paths);
     }
 
-    handleMultiPartContractVerification(bodyFormData, paths, hre.config.paths.root);
-
-    await axios
-        .post(voyagerUrl, bodyFormData.getBuffer(), {
-            headers: bodyFormData.getHeaders()
-        })
-        .catch(() => {
+    const sourceRegex = new RegExp("^" + hre.config.paths?.starknetSources + "/");
+    const contractNameDefault = mainPath.replace(sourceRegex, "");
+    // If contract name is not provided, use the default
+    bodyFormData.append("contract-name", args.contractName ? args.contractName : contractNameDefault);
```

Comment:
For a shorter solution, you can use the `||` operator as suggested in another comment.

Reply:
https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/120/commits/045f52ec8a3707b9f4a49750e1aa0c59a58772d7